### PR TITLE
Add option to spend badge points on wormholes

### PR DESCRIPTION
**Pull request by Xinayder**
Haven't tested it entirely, just the loop part. It works, but I'd recommend more testing.

### CODE REVIEW

**Comment by GreatOldOne** [comment]
buyInterval is out of scope for the second conditional check.

File: autoPlay.user.js [hunk 1]
```diff
@@ -411,6 +413,29 @@ function MainLoop() {
...
+				console.log('User has enough badge points to buy Wormholes, buying them.');
+				var buyInterval = w.setInterval(function() {
+					s().TrySpendBadgePoints(document.getElementById('purchase_abilityitem_26')); // todo: replace this function with something else
+					// update the dialog
+					s().m_UI.UpdateSpendBadgePointsDialog();
+					avalPoints = s().m_rgPlayerTechTree.badge_points;
+				}, 500);
+			}
+			
+			if (buyInterval) {
```
Comment (GreatOldOne) [inline autoPlay.user.js:434]: buyInterval is out of scope here.

File: autoPlay.user.js [hunk 2]
```diff
@@ -411,6 +413,29 @@ function MainLoop() {
...
 
 	var level = s().m_rgGameData.level + 1;
+	
+	// check if the spend badge points dialog is displayed
+	if (s().m_UI.m_spendBadgePointsDialog[0].style.display != 'none') {
+		if (enablePurchaseWormhole) {
+			var avalPoints = s().m_rgPlayerTechTree.badge_points;
+			var wormholeCost = 100;
+			
+			if (avalPoints >= wormholeCost) {
```
Comment (GreatOldOne) [inline autoPlay.user.js:424]: Must declare buyInterval before this check.

**Comment by Xinayder** [comment]
@GreatOldOne thanks for pointing out the issues. Gonna fix them

**Comment by tleunen** [comment]
This is already covered by #215

**Comment by rossengeorgiev** [comment]
This code PR has a lot of cruft. See discussion in https://github.com/SteamDatabase/steamSummerMinigame/pull/215#issuecomment-112807157

**Comment by Xinayder** [comment]
@rossengeorgiev @tleunen I made this PR because the other guy was doing auto-purchase badge points in general. This one is for wormholes only.

**Comment by rossengeorgiev** [comment]
closed in favor of #281

**Pull request closed without merging**